testing/logging/test_formatter.py

```diff
@@ -53,13 +53,80 @@ def test_multiline_message():
     # this is called by logging.Formatter.format
     record.message = record.getMessage()
 
-    style = PercentStyleMultiline(logfmt)
-    output = style.format(record)
+    ai_on_style = PercentStyleMultiline(logfmt, True)
+    output = ai_on_style.format(record)
     assert output == (
         "dummypath                   10 INFO     Test Message line1\n"
         "                                        line2"
     )
 
+    ai_off_style = PercentStyleMultiline(logfmt, False)
+    output = ai_off_style.format(record)
+    assert output == (
+        "dummypath                   10 INFO     Test Message line1\n" "line2"
+    )
+
+    ai_none_style = PercentStyleMultiline(logfmt, None)
+    output = ai_none_style.format(record)
+    assert output == (
+        "dummypath                   10 INFO     Test Message line1\n"
+        "                                        line2"
+    )
+
+    record.auto_indent = False
+    output = ai_on_style.format(record)
+    assert output == (
+        "dummypath                   10 INFO     Test Message line1\n" "line2"
```

Comment:
This is hard to read / confusing.
Please consider using `.splitlines()` instead and comparing lists instead.
Also below.

Reply:
If black gets in your way here you can also use `# fmt: off` … `# fmt: on`..

Reply:
This is the way the test was written before I introduced my feature change. I merely duplicated the check within the existing test, and adjusted it to check for the new behavior

Reply:
This is all new code, and I do not see `"…line1\n" "line2"` being used anywhere before?!

Reply:
lines 58-61

Reply:
They are not on the same line though.  That's what I mean.
So in your case `"…line1\nline2"` should be used at least.

Reply:
Also please squash and force-push it then already, since we cannot do this from the UI (#4361).

Reply:
I did not place them on the same line, the auto formatter did that. I'm fine with getting rid of the '" "', but once we've gotten to the point where we are code reviewing the auto formatter's contribution to test code, it starts to feel like code review for code review's sake? I've worked on this for over a week, and it would be really nice to get this merged

Reply:
I did not place them on the same line, the auto formatter did that. I'm fine with getting rid of the '" "', but once we've gotten to the point where we are code reviewing the auto formatter's contribution to test code, it starts to feel like code review for code review's sake? I've worked on this for over a week, and it would be really nice to get this merge

Reply:
> I did not place them on the same line, the auto formatter did that.

You ran `black` however (either manually or via pre-commit).
It is not smart enough in this case, so it makes sense to manually review it.

> I've worked on this for over a week, and it would be really nice to get this merged

Sure, but you are not the one maintaining it then from there likely.

I've only skimmed this in general.  As for me I do not like the added complexity, but can see that it is necessary.

Please squash / force-push - assuming that this is now ready from your side, and was approved already.

Reply:
Do you have the ability to merge the branch when codecov is still failing?

Reply:
Yes.

Reply:
squash/force-push complete

Reply:
With respect to your comment that you do not like the added complexity, I am in full agreement. It is my view that, while automatic indentation of multiline log entries may be desirable, it does not belong in the pytest project, it belongs in the logging project instead. That fact that it is here instead of there requires the addition of this extra logic to at least permit controlling/disabling the feature when desired.

Reply:
If this is ready to merge, please proceed, thx

Reply:
> That fact that it is here instead of there requires the addition of this extra logic to at least permit controlling/disabling the feature when desired.

Yes, unfortunately.
It might have been better to not enable it by default, but rather only add it via settings similar to what you've done now.
I've not looked into details, but could imagine that the "sane" behavior would be a good default then?

Reply:
I'd say the behavior is arguably sane. It just needed a bit of extra configuration to globally or selectively revert logging to the original behavior, plus threw in a little extra power because it was easy

Reply:
Is this ready to merge, then?

Reply:
> I'd say the behavior is arguably sane. It just needed a bit of extra configuration to globally or selectively revert logging to the original behavior

What I've meant is that it maybe good to change the defaults here, so that you would not have to configure anything - but people wanting the extra indent would have to instead.

Reply:
Personally, I agree the current behavior might not be a desirable default for some ... but I'd be wary of changing it out from under existing users.    For example: if people have done curious things with their ci logs, like parsing them and loading them into splunk, changing the default would cause those same people may have to modify their parsers.  I'm all for keeping the current default ... not because it's particularly more or less sane, but because leaving defaults the same is often a good choice.

Reply:
@earonesty
Thanks for the feedback.  I've thought we only changed this recently, but it is in 4.6.0 already (https://github.com/pytest-dev/pytest/commit/c8d23c206bd25d869acac2ee87d01289b04c3cda).

Reply:
But according to https://github.com/pytest-dev/pytest/pull/5312#issuecomment-498660973 it does not only affect the output, but also existing tests where `.text` is different then..! (not sure if that is addressed here when changing the output, and/or was fixed inbetween already).

Therefore (and in general) I'd be still +1 to restore the behavior before 4.6.0 by default (i.e. change the defaults in this PR).  It is unfortunate that it would change back and forth then, but seems to be better / more sane in the long term.
I assume @asottile agrees here (using pytest as a tool that should not do too much magic)?

@thisch 
What do you think?

Reply:
And as a justification of the restoration of the old default value, I would only say in the changelog that
"It is better to activate new features that may lead to broken tests explicitly instead of implicitly."